[Bug]: recurring transfer causes panic in market-sim fuzzing runs

### Problem encountered

vega-market-sim [PR](https://github.com/vegaprotocol/vega-market-sim/pull/526) narrows the fuzzing of recurring transfers so more transactions are accepted as valid.

When running fuzz test core panics:

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0x118 pc=0x103971b28]

goroutine 66254 [running]:
code.vegaprotocol.io/vega/core/execution/common.(*marketTracker).getRelativeReturnMetricTotal(...)
	/Users/caed/vega-market-sim/extern/vega/core/execution/common/market_activity_tracker.go:965
code.vegaprotocol.io/vega/core/execution/common.(*MarketActivityTracker).calculateMetricForParty(0x140014aad80, {0x1400c652340, 0x40}, {0x14003173040, 0x40}, {0x140069eaa30, 0x1, 0x1400cb73950?}, 0x6, 0x4)
	/Users/caed/vega-market-sim/extern/vega/core/execution/common/market_activity_tracker.go:708 +0x558
code.vegaprotocol.io/vega/core/execution/common.calculateMetricForTeamUtil({0x1400c652340, 0x40}, {0x1400e712a00?, 0x1, 0x1400e71b320?}, {0x140069eaa30, 0x1, 0x1}, 0x30?, 0x10636f320?, ...)
	/Users/caed/vega-market-sim/extern/vega/core/execution/common/market_activity_tracker.go:643 +0x16c
code.vegaprotocol.io/vega/core/execution/common.(*MarketActivityTracker).calculateMetricForTeam(0x105e7e400?, {0x1400c652340?, 0x14003173340?}, {0x1400e712a00?, 0x1400e54fb48?, 0x1028af4c8?}, {0x140069eaa30?, 0x0?, 0x1400e54fb68?}, 0x286594c?, ...)
	/Users/caed/vega-market-sim/extern/vega/core/execution/common/market_activity_tracker.go:624 +0x98
code.vegaprotocol.io/vega/core/execution/common.(*MarketActivityTracker).calculateMetricForTeams(0x140069cae70?, {0x1400c652340, 0x40}, 0x1400e71b2f0, {0x140069eaa30, 0x1, 0x1}, 0x5e645a0?, 0x1?, 0x1400e54fd08?, ...)
	/Users/caed/vega-market-sim/extern/vega/core/execution/common/market_activity_tracker.go:611 +0x298
code.vegaprotocol.io/vega/core/execution/common.(*MarketActivityTracker).CalculateMetricForTeams(0x140014aad80, 0x140069f07e0)
	/Users/caed/vega-market-sim/extern/vega/core/execution/common/market_activity_tracker.go:567 +0x14c
code.vegaprotocol.io/vega/core/banking.(*Engine).dispatchRequired(0x140005b6b60?, 0x1400e550090?)
	/Users/caed/vega-market-sim/extern/vega/core/banking/recurring_transfers.go:192 +0x74
code.vegaprotocol.io/vega/core/banking.(*Engine).distributeRecurringTransfers(0x140005b6b60, {0x106549c70, 0x1400e5406c0}, 0x12)
	/Users/caed/vega-market-sim/extern/vega/core/banking/recurring_transfers.go:288 +0x1894
code.vegaprotocol.io/vega/core/banking.(*Engine).OnEpoch(0x140005b6b60, {0x106549c70, 0x1400e5406c0}, {0x12, {0x0, 0xedbec18bc, 0x0}, {0x0, 0xedbec1934, 0x0}, ...})
	/Users/caed/vega-market-sim/extern/vega/core/banking/engine.go:279 +0xb4
code.vegaprotocol.io/vega/core/epochtime.(*Svc).notify(0x1400162a120, {0x106549c70, 0x1400e5406c0}, {0x12, {0x0, 0xedbec18bc, 0x0}, {0x0, 0xedbec1934, 0x0}, ...})
	/Users/caed/vega-market-sim/extern/vega/core/epochtime/service.go:204 +0xec
code.vegaprotocol.io/vega/core/epochtime.(*Svc).OnTick(0x1400162a120, {0x106549c70, 0x1400e5406c0}, {0x1?, 0x106564448?, 0x0?})
	/Users/caed/vega-market-sim/extern/vega/core/epochtime/service.go:134 +0x58c
code.vegaprotocol.io/vega/core/vegatime.(*Svc).notify(...)
	/Users/caed/vega-market-sim/extern/vega/core/vegatime/service.go:107
code.vegaprotocol.io/vega/core/vegatime.(*Svc).SetTimeNow(0x140014aac00, {0x106549c70, 0x1400e5406c0}, {0x40?, 0x105cccdc0?, 0x0?})
	/Users/caed/vega-market-sim/extern/vega/core/vegatime/service.go:80 +0x26c
code.vegaprotocol.io/vega/core/processor.(*App).OnBeginBlock(_, {{0x1400e3c3820, 0x20, 0x20}, {{0x0, 0x0}, {0x1400144f0b0, 0x6}, 0x897, {0x0, ...}, ...}, ...})
	/Users/caed/vega-market-sim/extern/vega/core/processor/abci.go:840 +0x5bc
code.vegaprotocol.io/vega/core/blockchain/abci.(*App).BeginBlock(_, {{0x1400e3c3820, 0x20, 0x20}, {{0x0, 0x0}, {0x1400144f0b0, 0x6}, 0x897, {0x0, ...}, ...}, ...})
	/Users/caed/vega-market-sim/extern/vega/core/blockchain/abci/abci.go:49 +0x64
code.vegaprotocol.io/vega/cmd/vega/commands/node.(*appW).BeginBlock(_, {{0x1400e3c3820, 0x20, 0x20}, {{0x0, 0x0}, {0x1400144f0b0, 0x6}, 0x897, {0x0, ...}, ...}, ...})
	/Users/caed/vega-market-sim/extern/vega/cmd/vega/commands/node/app_wrapper.go:67 +0x4c
code.vegaprotocol.io/vega/core/blockchain/nullchain.(*NullBlockchain).BeginBlock(0x14000274000)
	/Users/caed/vega-market-sim/extern/vega/core/blockchain/nullchain/nullchain.go:340 +0x2e4
code.vegaprotocol.io/vega/core/blockchain/nullchain.(*NullBlockchain).processBlock(0x14000274000)
	/Users/caed/vega-market-sim/extern/vega/core/blockchain/nullchain/nullchain.go:199 +0x1b8
code.vegaprotocol.io/vega/core/blockchain/nullchain.(*NullBlockchain).ForwardTime(0x14000274000, 0x3b9aca00)
	/Users/caed/vega-market-sim/extern/vega/core/blockchain/nullchain/nullchain.go:276 +0x198
code.vegaprotocol.io/vega/core/blockchain/nullchain.(*NullBlockchain).handleForwardTime.func1()
	/Users/caed/vega-market-sim/extern/vega/core/blockchain/nullchain/server.go:132 +0x2c
created by code.vegaprotocol.io/vega/core/blockchain/nullchain.(*NullBlockchain).handleForwardTime
	/Users/caed/vega-market-sim/extern/vega/core/blockchain/nullchain/server.go:131 +0x1ec
```

vega-market-sim dump includes logs and replay file:
[vega-sim-1lt2s6j2.zip](https://github.com/vegaprotocol/vega/files/12883166/vega-sim-1lt2s6j2.zip)



### Observed behaviour

Panic

### Expected behaviour

No panic

### Steps to reproduce

```bash
1. cd vega-market-sim
1. git checkout feat/rework_reward_fuzzing
1. python -m vega_sim.scenario.fuzzed_markets.run_fuzz_test -s 100

or

1. python -m vega_sim.replay.replay --dir PATH_TO_REPLAY_FILE

```


### Software version

8089a4cd795b13f8d8c3d5ed32b8a3afa85107b4

### Failing test

_No response_

### Jenkins run

_No response_

### Configuration used

```shell
{
  "genesis_time": "2023-05-09T10:00:00.000000000Z",
  "chain_id": "CUSTOM",
  "initial_height": "0",
  "consensus_params": {
    "block": {
      "max_bytes": "22020096",
      "max_gas": "-1",
      "time_iota_ms": "1"
    },
    "evidence": {
      "max_age_num_blocks": "100000",
      "max_age_duration": "172800000000000",
      "max_bytes": "1048576"
    },
    "validator": {
      "pub_key_types": ["ed25519"]
    },
    "version": {}
  },
  "validators": [
    {
      "address": "7DFD3744A773C166B36F170E4F5EF3E998703F4E",
      "pub_key": {
        "type": "tendermint/PubKeyEd25519",
        "value": "2lPO6I0p5uCBZzeoA2+ECJ2+57LtLVMjXpxH5za9TvE="
      },
      "power": "10",
      "name": ""
    }
  ],
  "app_hash": "",
  "app_state": {
    "assets": {
      "VOTE": {
        "name": "VOTE",
        "symbol": "VOTE",
        "decimals": 5,
        "min_lp_stake": "1",
        "source": {
          "builtin_asset": {
            "max_faucet_amount_mint": "100000000000"
          }
        }
      }
    },
    "validators": {
      "6NemqXSfN4zfZvAQeIANflEFqQ98izX9C0KLI2dITSY=": {
        "id": "c2b0ab69fb1bd19862c04df401920ea62d3e0bce41a50148240f42017824d17a",
        "vega_pub_key": "dfcf0fa8d67cfa11b3a511e320f3c47807a5e0735cc06dd9c2bc00f7ccc67117",
        "vega_pub_key_index": 1,
        "ethereum_address": "0x9e5BEEC6E56B28cCbd02864840B0f1e0125e42Ce",
        "tm_pub_key": "6NemqXSfN4zfZvAQeIANflEFqQ98izX9C0KLI2dITSY=",
        "info_url": "",
        "country": "",
        "name": "",
        "avatar_url": ""
      }
    },
    "network": {
      "replay_attack_threshold": 150
    },
    "network_parameters": {
      "blockchains.ethereumConfig": "{\"network_id\": \"3\", \"chain_id\": \"3\", \"collateral_bridge_contract\": { \"address\": \"0xa6F1E140daC13002Dfd9789D6dBA59117c717D7a\" }, \"confirmations\": 50, \"staking_bridge_contract\": { \"address\": \"0xfce2CC92203A266a9C8e67461ae5067c78f67235\", \"deployment_block_height\": 11001702}, \"multisig_control_contract\": {\"address\": \"0xCF6d41235911184fe6F35D47207813bFF3B91601\", \"deployment_block_height\": 12710009 } }",
      "governance.proposal.asset.maxClose": "8760h0m0s",
      "governance.proposal.asset.maxEnact": "8760h0m0s",
      "governance.proposal.asset.minClose": "1s",
      "governance.proposal.asset.minEnact": "1s",
      "governance.proposal.asset.minProposerBalance": "1",
      "governance.proposal.asset.minVoterBalance": "1",
      "governance.proposal.asset.requiredMajority": "0.66",
      "governance.proposal.asset.requiredParticipation": "0.00001",
      "governance.proposal.freeform.maxClose": "8760h0m0s",
      "governance.proposal.freeform.minClose": "1s",
      "governance.proposal.freeform.minProposerBalance": "1",
      "governance.proposal.freeform.minVoterBalance": "1",
      "governance.proposal.freeform.requiredMajority": "0.66",
      "governance.proposal.freeform.requiredParticipation": "0.00001",
      "governance.proposal.market.maxClose": "8760h0m0s",
      "governance.proposal.market.maxEnact": "8760h0m0s",
      "governance.proposal.market.minClose": "1s",
      "governance.proposal.market.minEnact": "2s",
      "governance.proposal.market.minProposerBalance": "1",
      "governance.proposal.market.minVoterBalance": "1",
      "governance.proposal.market.requiredMajority": "0.66",
      "governance.proposal.market.requiredParticipation": "0.00001",
      "governance.proposal.updateAsset.maxClose": "720h",
      "governance.proposal.updateAsset.maxEnact": "720h",
      "governance.proposal.updateAsset.minClose": "1s",
      "governance.proposal.updateAsset.minEnact": "1s",
      "governance.proposal.updateAsset.minProposerBalance": "1",
      "governance.proposal.updateAsset.minVoterBalance": "1",
      "governance.proposal.updateAsset.requiredMajority": "0.66",
      "governance.proposal.updateAsset.requiredParticipation": "0.09",
      "governance.proposal.updateMarket.maxClose": "8760h0m0s",
      "governance.proposal.updateMarket.maxEnact": "8760h0m0s",
      "governance.proposal.updateMarket.minClose": "1s",
      "governance.proposal.updateMarket.minEnact": "1s",
      "governance.proposal.updateMarket.minProposerBalance": "1",
      "governance.proposal.updateMarket.minProposerEquityLikeShare": "0",
      "governance.proposal.updateMarket.minVoterBalance": "1",
      "governance.proposal.updateMarket.requiredMajority": "0.66",
      "governance.proposal.updateMarket.requiredMajorityLP": "0.66",
      "governance.proposal.updateMarket.requiredParticipation": "0.00001",
      "governance.proposal.updateMarket.requiredParticipationLP": "0",
      "governance.proposal.updateNetParam.maxClose": "8760h0m0s",
      "governance.proposal.updateNetParam.maxEnact": "8760h0m0s",
      "governance.proposal.updateNetParam.minClose": "1s",
      "governance.proposal.updateNetParam.minEnact": "1s",
      "governance.proposal.updateNetParam.minProposerBalance": "1",
      "governance.proposal.updateNetParam.minVoterBalance": "1",
      "governance.proposal.updateNetParam.requiredMajority": "0.5",
      "governance.proposal.updateNetParam.requiredParticipation": "0.00001",
      "governance.proposal.referralProgram.maxClose": "8760h0m0s",
      "governance.proposal.referralProgram.maxEnact": "8760h0m0s",
      "governance.proposal.referralProgram.minClose": "1s",
      "governance.proposal.referralProgram.minEnact": "1s",
      "governance.proposal.referralProgram.minProposerBalance": "1",
      "governance.proposal.referralProgram.minVoterBalance": "1",
      "governance.proposal.referralProgram.requiredMajority": "0.5",
      "governance.proposal.referralProgram.requiredParticipation": "0.00001",
      "governance.proposal.VolumeDiscountProgram.maxClose": "8760h0m0s",
      "governance.proposal.VolumeDiscountProgram.maxEnact": "8760h0m0s",
      "governance.proposal.VolumeDiscountProgram.minClose": "1s",
      "governance.proposal.VolumeDiscountProgram.minEnact": "1s",
      "governance.proposal.VolumeDiscountProgram.minProposerBalance": "1",
      "governance.proposal.VolumeDiscountProgram.minVoterBalance": "1",
      "governance.proposal.VolumeDiscountProgram.requiredMajority": "0.5",
      "governance.proposal.VolumeDiscountProgram.requiredParticipation": "0.00001",
      "limits.markets.maxPeggedOrders": "1500",
      "market.auction.maximumDuration": "168h",
      "market.auction.minimumDuration": "1s",
      "market.fee.factors.infrastructureFee": "0.0005",
      "market.fee.factors.makerFee": "0.0002",
      "market.liquidity.bondPenaltyParameter": "0.1",
      "market.liquidity.earlyExitPenalty": "0.1",
      "market.liquidity.maximumLiquidityFeeFactorLevel": "0.03",
      "market.liquidity.minimum.probabilityOfTrading.lpOrders": "1e-6",
      "market.liquidity.probabilityOfTrading.tau.scaling": "10.0",
      "market.liquidity.providersFeeCalculationTimeStep": "1s",
      "market.liquidity.sla.nonPerformanceBondPenaltyMax": "0.5",
      "market.liquidity.sla.nonPerformanceBondPenaltySlope": "2",
      "market.liquidity.stakeToCcyVolume": "1.0",
      "market.liquidity.targetstake.triggering.ratio": "0.25",
      "market.liquidityProvision.minLpStakeQuantumMultiple": "5000",
      "market.liquidityProvision.shapes.maxSize": "100",
      "market.margin.scalingFactors": "{\"search_level\": 1.1, \"initial_margin\": 1.5, \"collateral_release\": 1.7}",
      "market.monitor.price.defaultParameters": "{\"triggers\": [{\"auction_extension\": 300, \"horizon\": 43200, \"probability\": \"0.9999999\"}] }",
      "market.stake.target.scalingFactor": "0.0001",
      "market.stake.target.timeWindow": "1h",
      "market.value.windowLength": "2h0m0s",
      "network.checkpoint.timeElapsedBetweenCheckpoints": "5m",
      "network.floatingPointUpdates.delay": "5m",
      "network.markPriceUpdateMaximumFrequency": "5s",
      "network.transaction.defaultgas": "1",
      "network.transactions.maxgasperblock": "3000000",
      "network.transactions.minBlockCapacity": "32",
      "network.validators.ersatz.multipleOfTendermintValidators": "0",
      "network.validators.ersatz.rewardFactor": "0.85",
      "network.validators.incumbentBonus": "0.05",
      "network.validators.minimumEthereumEventsForNewValidator": "3",
      "network.validators.multisig.numberOfSigners": "13",
      "network.validators.tendermint.number": "13",
      "reward.asset": "VOTE",
      "reward.staking.delegation.competitionLevel": "1.3",
      "reward.staking.delegation.delegatorShare": "0.8",
      "reward.staking.delegation.maxPayoutPerEpoch": "9890000000000000000000",
      "reward.staking.delegation.maxPayoutPerParticipant": "700000000000000000000",
      "reward.staking.delegation.minimumValidatorStake": "3000000000000000000000",
      "reward.staking.delegation.minValidators": "5",
      "reward.staking.delegation.optimalStakeMultiplier": "3.0",
      "reward.staking.delegation.payoutDelay": "0h",
      "reward.staking.delegation.payoutFraction": "1.0",
      "rewards.marketCreationQuantumMultiple": "50000000",
      "snapshot.interval.length": "300",
      "spam.pow.difficulty": "15",
      "spam.pow.hashFunction": "sha3_24_rounds",
      "spam.pow.increaseDifficulty": "0",
      "spam.pow.numberOfPastBlocks": "100",
      "spam.pow.numberOfTxPerBlock": "1000",
      "spam.protection.delegation.min.tokens": "100000000000000000",
      "spam.protection.max.batchSize": "30",
      "spam.protection.max.delegations": "360",
      "spam.protection.max.proposals": "300",
      "spam.protection.max.votes": "300",
      "spam.protection.maxUserTransfersPerEpoch": "10000",
      "spam.protection.minimumWithdrawalQuantumMultiple": "10",
      "spam.protection.minMultisigUpdates": "100000000000000000000",
      "spam.protection.proposal.min.tokens": "2000000000000000000000",
      "spam.protection.voting.min.tokens": "1000000000000000000",
      "spam.protection.applyReferral.min.funds": "0",
      "transfer.fee.factor": "0.001",
      "transfer.minTransferQuantumMultiple": "100",
      "validator.performance.scaling.factor": "0",
      "validators.delegation.minAmount": "1",
      "validators.epoch.length": "2m",
      "validators.vote.required": "0.67",
      "referralProgram.maxReferralTiers": "10",
      "referralProgram.maxReferralRewardFactor": "1",
      "referralProgram.maxReferralDiscountFactor": "1",
      "referralProgram.maxPartyNotionalVolumeByQuantumPerEpoch": "10000",
      "referralProgram.minStakedVegaTokens": "0",
      "referralProgram.maxReferralRewardProportion": "1"
    },
    "network_limits": {
      "propose_market_enabled": true,
      "propose_asset_enabled": true,
      "bootstrap_block_count": 0
    },
    "checkpoint": {
      "load_hash": ""
    }
  }
}
```


### Relevant log output

```shell
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0x118 pc=0x103971b28]

goroutine 66254 [running]:
code.vegaprotocol.io/vega/core/execution/common.(*marketTracker).getRelativeReturnMetricTotal(...)
	/Users/caed/vega-market-sim/extern/vega/core/execution/common/market_activity_tracker.go:965
code.vegaprotocol.io/vega/core/execution/common.(*MarketActivityTracker).calculateMetricForParty(0x140014aad80, {0x1400c652340, 0x40}, {0x14003173040, 0x40}, {0x140069eaa30, 0x1, 0x1400cb73950?}, 0x6, 0x4)
	/Users/caed/vega-market-sim/extern/vega/core/execution/common/market_activity_tracker.go:708 +0x558
code.vegaprotocol.io/vega/core/execution/common.calculateMetricForTeamUtil({0x1400c652340, 0x40}, {0x1400e712a00?, 0x1, 0x1400e71b320?}, {0x140069eaa30, 0x1, 0x1}, 0x30?, 0x10636f320?, ...)
	/Users/caed/vega-market-sim/extern/vega/core/execution/common/market_activity_tracker.go:643 +0x16c
code.vegaprotocol.io/vega/core/execution/common.(*MarketActivityTracker).calculateMetricForTeam(0x105e7e400?, {0x1400c652340?, 0x14003173340?}, {0x1400e712a00?, 0x1400e54fb48?, 0x1028af4c8?}, {0x140069eaa30?, 0x0?, 0x1400e54fb68?}, 0x286594c?, ...)
	/Users/caed/vega-market-sim/extern/vega/core/execution/common/market_activity_tracker.go:624 +0x98
code.vegaprotocol.io/vega/core/execution/common.(*MarketActivityTracker).calculateMetricForTeams(0x140069cae70?, {0x1400c652340, 0x40}, 0x1400e71b2f0, {0x140069eaa30, 0x1, 0x1}, 0x5e645a0?, 0x1?, 0x1400e54fd08?, ...)
	/Users/caed/vega-market-sim/extern/vega/core/execution/common/market_activity_tracker.go:611 +0x298
code.vegaprotocol.io/vega/core/execution/common.(*MarketActivityTracker).CalculateMetricForTeams(0x140014aad80, 0x140069f07e0)
	/Users/caed/vega-market-sim/extern/vega/core/execution/common/market_activity_tracker.go:567 +0x14c
code.vegaprotocol.io/vega/core/banking.(*Engine).dispatchRequired(0x140005b6b60?, 0x1400e550090?)
	/Users/caed/vega-market-sim/extern/vega/core/banking/recurring_transfers.go:192 +0x74
code.vegaprotocol.io/vega/core/banking.(*Engine).distributeRecurringTransfers(0x140005b6b60, {0x106549c70, 0x1400e5406c0}, 0x12)
	/Users/caed/vega-market-sim/extern/vega/core/banking/recurring_transfers.go:288 +0x1894
code.vegaprotocol.io/vega/core/banking.(*Engine).OnEpoch(0x140005b6b60, {0x106549c70, 0x1400e5406c0}, {0x12, {0x0, 0xedbec18bc, 0x0}, {0x0, 0xedbec1934, 0x0}, ...})
	/Users/caed/vega-market-sim/extern/vega/core/banking/engine.go:279 +0xb4
code.vegaprotocol.io/vega/core/epochtime.(*Svc).notify(0x1400162a120, {0x106549c70, 0x1400e5406c0}, {0x12, {0x0, 0xedbec18bc, 0x0}, {0x0, 0xedbec1934, 0x0}, ...})
	/Users/caed/vega-market-sim/extern/vega/core/epochtime/service.go:204 +0xec
code.vegaprotocol.io/vega/core/epochtime.(*Svc).OnTick(0x1400162a120, {0x106549c70, 0x1400e5406c0}, {0x1?, 0x106564448?, 0x0?})
	/Users/caed/vega-market-sim/extern/vega/core/epochtime/service.go:134 +0x58c
code.vegaprotocol.io/vega/core/vegatime.(*Svc).notify(...)
	/Users/caed/vega-market-sim/extern/vega/core/vegatime/service.go:107
code.vegaprotocol.io/vega/core/vegatime.(*Svc).SetTimeNow(0x140014aac00, {0x106549c70, 0x1400e5406c0}, {0x40?, 0x105cccdc0?, 0x0?})
	/Users/caed/vega-market-sim/extern/vega/core/vegatime/service.go:80 +0x26c
code.vegaprotocol.io/vega/core/processor.(*App).OnBeginBlock(_, {{0x1400e3c3820, 0x20, 0x20}, {{0x0, 0x0}, {0x1400144f0b0, 0x6}, 0x897, {0x0, ...}, ...}, ...})
	/Users/caed/vega-market-sim/extern/vega/core/processor/abci.go:840 +0x5bc
code.vegaprotocol.io/vega/core/blockchain/abci.(*App).BeginBlock(_, {{0x1400e3c3820, 0x20, 0x20}, {{0x0, 0x0}, {0x1400144f0b0, 0x6}, 0x897, {0x0, ...}, ...}, ...})
	/Users/caed/vega-market-sim/extern/vega/core/blockchain/abci/abci.go:49 +0x64
code.vegaprotocol.io/vega/cmd/vega/commands/node.(*appW).BeginBlock(_, {{0x1400e3c3820, 0x20, 0x20}, {{0x0, 0x0}, {0x1400144f0b0, 0x6}, 0x897, {0x0, ...}, ...}, ...})
	/Users/caed/vega-market-sim/extern/vega/cmd/vega/commands/node/app_wrapper.go:67 +0x4c
code.vegaprotocol.io/vega/core/blockchain/nullchain.(*NullBlockchain).BeginBlock(0x14000274000)
	/Users/caed/vega-market-sim/extern/vega/core/blockchain/nullchain/nullchain.go:340 +0x2e4
code.vegaprotocol.io/vega/core/blockchain/nullchain.(*NullBlockchain).processBlock(0x14000274000)
	/Users/caed/vega-market-sim/extern/vega/core/blockchain/nullchain/nullchain.go:199 +0x1b8
code.vegaprotocol.io/vega/core/blockchain/nullchain.(*NullBlockchain).ForwardTime(0x14000274000, 0x3b9aca00)
	/Users/caed/vega-market-sim/extern/vega/core/blockchain/nullchain/nullchain.go:276 +0x198
code.vegaprotocol.io/vega/core/blockchain/nullchain.(*NullBlockchain).handleForwardTime.func1()
	/Users/caed/vega-market-sim/extern/vega/core/blockchain/nullchain/server.go:132 +0x2c
created by code.vegaprotocol.io/vega/core/blockchain/nullchain.(*NullBlockchain).handleForwardTime
	/Users/caed/vega-market-sim/extern/vega/core/blockchain/nullchain/server.go:131 +0x1ec
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -189,6 +189,7 @@
 - [8364](https://github.com/vegaprotocol/vega/issues/8364) - Initialising from network history not working after database wipe
 - [8827](https://github.com/vegaprotocol/vega/issues/8827) - Add block height validation to validator initiated transactions and pruning to the `pow` engine cache
 - [8836](https://github.com/vegaprotocol/vega/issues/8836) - Fix enactment of market update state
+- [9760](https://github.com/vegaprotocol/vega/issues/9760) - Validate that a recurring transfer with markets matches the asset in the transfer.
 - [8848](https://github.com/vegaprotocol/vega/issues/8848) - Handle the case where the market is terminated and the epoch ends at the same block.
 - [8853](https://github.com/vegaprotocol/vega/issues/8853) - Liquidity provision amendment bug fixes
 - [8862](https://github.com/vegaprotocol/vega/issues/8862) - Fix settlement via governance

--- a/core/banking/engine.go
+++ b/core/banking/engine.go
@@ -111,6 +111,7 @@ type MarketActivityTracker interface {
 	CalculateMetricForTeams(ds *vega.DispatchStrategy) ([]*types.PartyContributionScore, map[string][]*types.PartyContributionScore)
 	GetMarketsWithEligibleProposer(asset string, markets []string, payoutAsset string, funder string) []*types.MarketContributionScore
 	MarkPaidProposer(asset, market, payoutAsset string, marketsInScope []string, funder string)
+	MarketTrackedForAsset(market, asset string) bool
 }
 
 type EthereumEventSource interface {

--- a/core/banking/mocks/mocks.go
+++ b/core/banking/mocks/mocks.go
@@ -540,6 +540,20 @@ func (mr *MockMarketActivityTrackerMockRecorder) MarkPaidProposer(arg0, arg1, ar
 	return mr.mock.ctrl.RecordCallWithMethodType(mr.mock, "MarkPaidProposer", reflect.TypeOf((*MockMarketActivityTracker)(nil).MarkPaidProposer), arg0, arg1, arg2, arg3, arg4)
 }
 
+// MarketTrackedForAsset mocks base method.
+func (m *MockMarketActivityTracker) MarketTrackedForAsset(arg0, arg1 string) bool {
+	m.ctrl.T.Helper()
+	ret := m.ctrl.Call(m, "MarketTrackedForAsset", arg0, arg1)
+	ret0, _ := ret[0].(bool)
+	return ret0
+}
+
+// MarketTrackedForAsset indicates an expected call of MarketTrackedForAsset.
+func (mr *MockMarketActivityTrackerMockRecorder) MarketTrackedForAsset(arg0, arg1 interface{}) *gomock.Call {
+	mr.mock.ctrl.T.Helper()
+	return mr.mock.ctrl.RecordCallWithMethodType(mr.mock, "MarketTrackedForAsset", reflect.TypeOf((*MockMarketActivityTracker)(nil).MarketTrackedForAsset), arg0, arg1)
+}
+
 // MockERC20BridgeView is a mock of ERC20BridgeView interface.
 type MockERC20BridgeView struct {
 	ctrl     *gomock.Controller

--- a/core/banking/recurring_transfers.go
+++ b/core/banking/recurring_transfers.go
@@ -66,6 +66,20 @@ func (e *Engine) recurringTransfer(
 				return fmt.Errorf("could not transfer funds, invalid asset for metric: %w", err)
 			}
 		}
+
+		if hasAsset && len(transfer.DispatchStrategy.Markets) > 0 {
+			asset := transfer.DispatchStrategy.AssetForMetric
+			for _, mid := range transfer.DispatchStrategy.Markets {
+				if !e.marketActivityTracker.MarketTrackedForAsset(mid, asset) {
+					transfer.Status = types.TransferStatusRejected
+					e.log.Debug("cannot transfer funds, invalid market for dispatch asset",
+						logging.String("mid", mid),
+						logging.String("asset", asset),
+					)
+					return errors.New("could not transfer funds, invalid market for dispatch asset")
+				}
+			}
+		}
 	}
 
 	if err := transfer.IsValid(); err != nil {

--- a/core/banking/recurring_transfers_test.go
+++ b/core/banking/recurring_transfers_test.go
@@ -715,3 +715,49 @@ func testRecurringTransferInvalidTransfers(t *testing.T) {
 		)
 	})
 }
+
+func TestMarketAssetMismatchRejectsTransfer(t *testing.T) {
+	eng := getTestEngine(t)
+
+	fromAcc := types.Account{
+		Balance: num.NewUint(1000),
+	}
+
+	eng.assets.EXPECT().Get(gomock.Any()).AnyTimes().Return(assets.NewAsset(&mockAsset{num.DecimalFromFloat(100)}), nil)
+	eng.tsvc.EXPECT().GetTimeNow().Times(1)
+	eng.col.EXPECT().GetPartyGeneralAccount(gomock.Any(), gomock.Any()).AnyTimes().Return(&fromAcc, nil)
+	eng.broker.EXPECT().Send(gomock.Any()).AnyTimes()
+	eng.col.EXPECT().TransferFunds(gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any()).AnyTimes()
+
+	recurring := &types.TransferFunds{
+		Kind: types.TransferCommandKindRecurring,
+		Recurring: &types.RecurringTransfer{
+			TransferBase: &types.TransferBase{
+				From:            "03ae90688632c649c4beab6040ff5bd04dbde8efbf737d8673bbda792a110301",
+				FromAccountType: types.AccountTypeGeneral,
+				To:              "2e05fd230f3c9f4eaf0bdc5bfb7ca0c9d00278afc44637aab60da76653d7ccf0",
+				ToAccountType:   types.AccountTypeGeneral,
+				Asset:           "eth",
+				Amount:          num.NewUint(10),
+				Reference:       "someref",
+			},
+			StartEpoch: 10,
+			EndEpoch:   nil, // forever
+			Factor:     num.MustDecimalFromString("0.9"),
+			DispatchStrategy: &vega.DispatchStrategy{
+				AssetForMetric:       "zohar",
+				Metric:               vega.DispatchMetric_DISPATCH_METRIC_AVERAGE_POSITION,
+				Markets:              []string{"mmm"},
+				EntityScope:          vega.EntityScope_ENTITY_SCOPE_INDIVIDUALS,
+				IndividualScope:      vega.IndividualScope_INDIVIDUAL_SCOPE_IN_TEAM,
+				WindowLength:         1,
+				LockPeriod:           1,
+				DistributionStrategy: vega.DistributionStrategy_DISTRIBUTION_STRATEGY_RANK,
+			},
+		},
+	}
+
+	// if in-scope market has a different asset it is rejected
+	eng.marketActivityTracker.EXPECT().MarketTrackedForAsset(gomock.Any(), gomock.Any()).Times(1).Return(false)
+	require.Error(t, eng.TransferFunds(context.Background(), recurring))
+}

--- a/core/banking/snapshot_test.go
+++ b/core/banking/snapshot_test.go
@@ -533,6 +533,7 @@ func TestRecurringTransfersSnapshotRoundTrip(t *testing.T) {
 	p, _ := proto.Marshal(recurring.Recurring.DispatchStrategy)
 	dsHash := hex.EncodeToString(crypto.Hash(p))
 
+	eng.marketActivityTracker.EXPECT().MarketTrackedForAsset("mmm", "zohar").Times(1).Return(true)
 	require.NoError(t, eng.TransferFunds(ctx, recurring))
 
 	// test the new transfer prompts a change

--- a/core/execution/common/market_activity_tracker.go
+++ b/core/execution/common/market_activity_tracker.go
@@ -353,6 +353,16 @@ func (mat *MarketActivityTracker) GetAllMarketIDs() []string {
 	return mIDs
 }
 
+// MarketTrackedForAsset returns whether the given market is seen to have the given asset by the tracker.
+func (mat *MarketActivityTracker) MarketTrackedForAsset(market, asset string) bool {
+	if markets, ok := mat.assetToMarketTrackers[asset]; ok {
+		if _, ok = markets[market]; ok {
+			return true
+		}
+	}
+	return false
+}
+
 // removeMarket is called when the market is removed from the network. It is not immediately removed to give a chance for rewards to be paid at the end of the epoch for activity during the epoch.
 // Instead it is marked for removal and will be removed at the beginning of the next epoch.
 func (mat *MarketActivityTracker) RemoveMarket(asset, marketID string) {

--- a/core/execution/common/market_activity_tracker_test.go
+++ b/core/execution/common/market_activity_tracker_test.go
@@ -66,6 +66,9 @@ func TestMarketTracker(t *testing.T) {
 	tracker.MarketProposed("asset1", "market1", "me")
 	tracker.MarketProposed("asset1", "market2", "me2")
 
+	assert.True(t, tracker.MarketTrackedForAsset("market1", "asset1"))
+	assert.False(t, tracker.MarketTrackedForAsset("market1", "asset2"))
+
 	require.Equal(t, false, tracker.IsMarketEligibleForBonus("asset1", "market1", "VEGA", []string{}, "zohar"))
 	require.Equal(t, false, tracker.IsMarketEligibleForBonus("asset1", "market2", "VEGA", []string{}, "zohar"))
 

